commit 29ea5662df8555408f7afc66766eb9bd55366395 (refs/changes/31/1231/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-01-10T12:13:15-08:00 (2 years, 9 months ago)
    
    joyent/node-cueball#69 connectionFSM could log when it's trying to connect, abandoning, and delaying

diff --git a/lib/connection-fsm.js b/lib/connection-fsm.js
index bcbd5a1..2f8843e 100644
--- a/lib/connection-fsm.js
+++ b/lib/connection-fsm.js
@@ -325,6 +325,9 @@ ConnectionFSM.prototype.state_error = function (S) {
 		--this.cf_retriesLeft;
 
 	if (this.cf_retries === Infinity || this.cf_retriesLeft > 0) {
+		this.cf_log.trace(this.cf_lastError, 'failed to connect to ' +
+		    'backend %s (%j), delaying before retry',
+		    this.cf_backend.key, this.cf_backend);
 		S.gotoState('delay');
 	} else {
 		this.cf_log.warn(this.cf_lastError, 'failed to connect to ' +
@@ -611,6 +614,9 @@ ConnectionFSM.prototype.state_busy = function (S) {
 		}
 	});
 	S.on(this.cf_conn, 'error', function (err) {
+		self.cf_log.error(err, 'connection to %s (%j) emitted ' +
+		    '"error" while busy (claimed)', this.cf_backend.key,
+		    this.cf_backend);
 		self.cf_lastError = err;
 		S.gotoState('error');
 		self.cf_pool._incrCounter('error-during-busy');
diff --git a/package.json b/package.json
index c1000b6..151ef20 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "1.3.0",
+  "version": "1.3.1",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
